x-pack/plugins/security/public/views/nav_control/nav_control.js

```diff
@@ -47,14 +47,14 @@ module.controller('securityNavController', ($scope, ShieldUser, globalNavState,
 });
 
 
-chromeHeaderNavControlsRegistry.register((ShieldUser, kbnBaseUrl, Private) => ({
+chromeHeaderNavControlsRegistry.register((ShieldUser, kbnBaseUrl, Private, showUserProfile) => ({
   name: 'security',
   order: 1000,
   side: NavControlSide.Right,
   render(el) {
     const xpackInfo = Private(XPackInfoProvider);
     const showSecurityLinks = xpackInfo.get('features.security.showLinks');
-    if (Private(PathProvider).isLoginOrLogout() || !showSecurityLinks) return null;
+    if (Private(PathProvider).isLoginOrLogout() || !showSecurityLinks || !showUserProfile) return null;
```

Comment:
Ah, good catch. I think `showSecurityLinks` ends up being undefined for unauthenticated routes because we can't load xpack license information. This line will get cleaned up once `isLogonOrLogout` is renamed anyway, and I think that will be powerful enough to allow us to remove `showUserProfile` altogether 👍